src/WebJobs.Script/Binding/Http/HttpTriggerAttributeBindingProvider.cs

```diff
@@ -125,20 +131,60 @@ public Type TriggerValueType
                     throw new NotSupportedException("An HttpRequestMessage is required");
                 }
 
+                HttpResponseMessage responseObjectInFunction = null;
+                if (_isProxy)
+                {
+                    string content = null;
+                    if (request.Content != null)
+                    {
+                        content = await request.Content.ReadAsStringAsync();
+                    }
+
+                    if (_parameter.ParameterType == typeof(HttpResponseMessage))
+                    {
+                        responseObjectInFunction = new HttpResponseMessage();
+
+                        if (!string.IsNullOrEmpty(content))
+                        {
+                            responseObjectInFunction.Deserialize(content);
+                        }
+                    }
+                    else if(_parameter.ParameterType == typeof(HttpRequestMessage))
+                    {
+                        var requestObjectInFunction = new HttpRequestMessage();
+
+                        if (!string.IsNullOrEmpty(content))
+                        {
+                            requestObjectInFunction.Deserialize(content);
+                        }
+
+                        // Adding the original request object to the newly created request object's properties as this will be needed when returning response to the client.
+                        requestObjectInFunction.Properties[ScriptConstants.AzureFunctionsHttpProxyRoutingDataKey] = request;
+                        request = requestObjectInFunction;
+                    }
+                }
+
                 IValueProvider valueProvider = null;
                 object poco = null;
                 IReadOnlyDictionary<string, object> userTypeBindingData = null;
                 string invokeString = ToInvokeString(request);
                 if (_isUserTypeBinding)
                 {
-                    valueProvider = await CreateUserTypeValueProvider(request, invokeString);
-                    if (_bindingDataProvider != null)
+                    if (_parameter.ParameterType == typeof(HttpResponseMessage))
```

Comment:
This logic is going to run for non proxy requests as well - I don't think we want that? I'm trying to avoid breaking changes here. What is this doing exactly?